src/CTA.Rules.Actions/ActionHelpers/ConfigMigrate.cs

```diff
@@ -38,25 +41,26 @@ private string MigrateWebConfig()
             var configXml = LoadWebConfig(_projectDir);
             if (configXml == null) { return isConfigFound; }
 
-            var config = ProcessWebConfig(configXml);
-            if (!string.IsNullOrEmpty(config))
+            var config = ProcessWebConfig(configXml, TemplateHelper.GetTemplateFileContent("", _projectType, Constants.appSettingsJson));
+            if (_hasData)
             {
                 isConfigFound = "Found and migrated settings from web.config";
-                AddAppSettingsJsonFile(config, TemplateHelper.GetTemplateFileContent("", _projectType, Constants.appSettingsJson), _projectDir);
+                AddAppSettingsJsonFile(config, _projectDir);
             }
 
             return isConfigFound;
         }
-        private XDocument LoadWebConfig(string projectDir)
+        private Configuration LoadWebConfig(string projectDir)
         {
             string webConfigFile = Path.Combine(projectDir, Constants.webConfig);
 
             if (File.Exists(webConfigFile))
             {
                 try
                 {
-                    var webConfig = XDocument.Load(webConfigFile);
-                    return webConfig;
+                    var fileMap = new ExeConfigurationFileMap() { ExeConfigFilename = webConfigFile };
+                    var configuration = ConfigurationManager.OpenMappedExeConfiguration(fileMap, ConfigurationUserLevel.None);
+                    return configuration;
                 }
                 catch (Exception ex)
                 {
```

Comment:
are we capturing when reading configs fail in some form of metrics ? Or if we have a way to do in ELS/KIBANA we should be ok too...

Reply:
Yes, failing to read config or write to settings file is part of our metrics.